src/sentry/preprod/size_analysis/utils.py

```diff
@@ -27,18 +36,48 @@ def can_compare_size_metrics(
                 "base_metric_length": len(base_metric),
             },
         )
-        return False
+        return ComparisonValidationResult(
+            can_compare=False,
+            error_message=f"Head and base have different numbers of size metrics. Head has {len(head_metrics)} metric(s), base has {len(base_metric)} metric(s).",
+        )
 
     # Build sets of (metrics_artifact_type, identifier) for both lists
     head_set = {(m.metrics_artifact_type, m.identifier) for m in head_metrics}
     base_set = {(m.metrics_artifact_type, m.identifier) for m in base_metric}
 
     # Return True if the sets are equal (all metrics match on both dimensions)
     if head_set == base_set:
-        return True
+        return ComparisonValidationResult(can_compare=True)
+
+    # Find the differences to provide detailed error message
+    head_only = head_set - base_set
+    base_only = base_set - head_set
+
+    def format_metric_key(key: tuple[int | None, str | None]) -> str:
+        artifact_type, identifier = key
+        type_name = "Unknown"
+        if artifact_type is not None:
+            try:
+                type_name = PreprodArtifactSizeMetrics.MetricsArtifactType(artifact_type).name
+            except (ValueError, AttributeError):
+                type_name = f"Type({artifact_type})"
+        identifier_str = identifier if identifier else "(no identifier)"
+        return f"{type_name}: {identifier_str}"
+
+    error_parts = ["Head and base size metrics cannot be compared due to mismatched metrics."]
+
+    if head_only:
+        formatted_head = [format_metric_key(k) for k in head_only]
+        error_parts.append(f"Head has metric(s) not in base: {', '.join(formatted_head)}")
```

Comment:
This isn't the most user-friendly message. Open to ideas on how to improve it.

Reply:
Yea I don't think users will understand how we've structured our data as "size metrics". Maybe for this case focus on the app_id? E.g. "`com.hackernews` not found in base build"

Reply:
Yeah, that's a good point. So I added specific messages for the more specific cases but if none of those work then we'll use this generic messages.